host/crypto/openssl/key.c

```diff
@@ -41,6 +42,85 @@ void oe_private_key_init(
     impl->pkey = pkey;
 }
 
+oe_result_t oe_private_key_from_engine(
+    const char *engine_id,
+    const char *engine_load_path,
+    const char *key_id,
+    oe_private_key_t* key,
+    int key_type,
+    uint64_t magic)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    oe_private_key_t* impl = (oe_private_key_t*)key;
```

Comment:
why is a cast here?  key and impl are both defined to be of the same type.

Reply:
done